### PR TITLE
Try linters

### PR DESCRIPTION
**Pull request by sam-github**
/to @kraman @rmg 

It tried a pass through strong-supervisor using eslint, jscs, and jshint.
I started from loopback modules dot-files for three linters.

Summary:

jshint caught only a single unnecessary semi-colon over eslint and jscs, and has the largest number of false positives that I can't find a way to configure away.

thumbs down on jshint

eslint actually founds some outright bugs, two thumbs up!

Though it has an un-workable bug in that it doesn't support node modules... it hates the top-level return, but this might be fixed on its master: https://github.com/eslint/eslint/issues/1158#issuecomment-75142625

jscs did pretty well, and is pretty customizable. It found a bunch of style things eslint didn't, perhaps because I hadn't turned enough things on in .eslintrc? Not sure, but it only had one false-positive, I like it for that, and I can work-around that.

I think we should add support for eslint and jscs, and also add support to strong-tools to provision a git pre-commit hook to run them (along with the standard git whitespace check). The git pre-commit hook has an example of only running across changed files, so we could even fix code up as we change it.

What do you think?

### CODE REVIEW

**Comment by rmg** [comment]
@sam-github it would also be nice if someone updated our build-framework scripts to [support more than just jshint](https://github.com/strongloop-internal/build-framework/blob/master/lib/npm/package.rb#L248-L268), probably selecting the appropriate one based on which is listed as a dev dep.

**Comment by kraman** [comment]
I like eslint and have been using it for a while on my own projects. The rules can also be customized in code and adding new rules seems easy to do. Haven't had a look at jscs yet.

**Comment by sam-github** [comment]
I'm running into some eslint issues, I'll merge this once they are sorted out, then the great "which .eslintrc file should we all use" bike-shedding can begin. :-()

**Comment by sam-github** [comment]
@rmg @kraman PTAL

Note a few things:

- eslint and jscs are dev deps
- eslint depends on github, because it needs a bug fix that isn't released yet
- each linter is a single npm script.. seems like it will be easier for CI, but I didn't select CI friend output formats, perhaps I should?
- I hate warnings. It passes or not, I don't want to have to evaluate "maybe OKs" every time the tool runs, so I made everything an error, or turned it off

**Comment by rmg** [comment]
@sam-github this PR broke the build. This PR was failing before merge, and now master is failing on the same tests:
```
supervisor --detach.creates a detached process
supervisor --detach.creates a log file matching spec
```

File: lib/config.js [hunk 1]
```diff
@@ -199,12 +205,13 @@ config.start = function start() {
...
-                                              { id: 0,
-                                                pid: process.pid });
+      var supervisorLogName = generateLogName(options.log, {
+        id: 0,
+        pid: process.pid
+      });
       process.env.supervisor_detached = true;
       var child = detach(options.argv.concat(options.args), supervisorLogName);
-      console.error('supervisor %d detached process %d, output logged to \'%s\'',
+      console.error('supervisor %d detached process %d, output log in \'%s\'',
```
Comment (rmg) [inline lib/config.js]: test/supervisor-detach.js depends on this log message so those tests now fail.

**Comment by rmg** [comment]
@sam-github I fixed the test failure on master directly.

**Pull request merged**